distribution/lib/Standard/AWS/0.0.0-dev/src/S3/S3_Data_Link.enso

```diff
@@ -0,0 +1,41 @@
+from Standard.Base import all
+import Standard.Base.Errors.Illegal_State.Illegal_State
+from Standard.Base.Enso_Cloud.Public_Utils import get_required_field
+from Standard.Base.Enso_Cloud.Data_Link import parse_format, parse_secure_value
+
+import project.AWS_Credential.AWS_Credential
+import project.S3.S3_File.S3_File
+
+## PRIVATE
+type S3_Data_Link
+    ## PRIVATE
+    Value (uri : Text) format (credentials : AWS_Credential | Nothing)
+
+    ## PRIVATE
+    parse json -> S3_Data_Link =
+        uri = get_required_field "uri" json
+        auth = parse_aws_auth (get_required_field "auth" json)
+        format = parse_format (json.get "format" Nothing)
+        S3_Data_Link.Value uri format auth
+
+    ## PRIVATE
+    as_file self -> S3_File = S3_File.new self.uri self.credentials
+
+    ## PRIVATE
+    read self (on_problems : Problem_Behavior) =
+        self.as_file.read self.format on_problems
+
+## PRIVATE
+parse_aws_auth json -> AWS_Credential | Nothing =
```

Comment:
possibly belongs in `AWS_Credential` but we can move when we have more data links for AWS.

Reply:
Moved to `Internal.Data_Link_Helpers`